db/compaction/compaction_picker_universal.cc

```diff
@@ -811,8 +814,18 @@ Compaction* UniversalCompactionBuilder::PickCompactionToReduceSizeAmp() {
         cf_name_.c_str(), file_num_buf, start_index, " to reduce size amp.\n");
   }
 
+  // size of the base sorted run for size amp calculation
+  uint64_t base_sr_size = sorted_runs_.back().size;
+  size_t sr_end_idx = sorted_runs_.size() - 1;
+  if (ioptions_.preclude_last_level_data_seconds > 0) {
+    if (sorted_runs_.back().level == ioptions_.num_levels - 1) {
+      sr_end_idx = sorted_runs_.size() - 2;
```

Comment:
Do we need to do an array index boundary check?

Reply:
Yes, added.